application/ui/src/features/inspect/train-model/train-model-device-picker.component.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useMemo } from 'react';
+
+import { $api } from '@geti-inspect/api';
+import { Button, Content, Flex, Heading, InlineAlert, Loading, Radio, RadioGroup, Text } from '@geti/ui';
+
+import { getDeviceMetadata } from './utils/device-metadata';
+
+import classes from './train-model.module.scss';
+
+interface TrainModelDevicePickerProps {
+    selectedDevice: string | null;
+    onSelect: (device: string | null) => void;
+}
+
+export const TrainModelDevicePicker = ({ selectedDevice, onSelect }: TrainModelDevicePickerProps) => {
+    const {
+        data: availableDevices,
+        isLoading,
+        isError,
+        isRefetching,
+        refetch,
+    } = $api.useQuery('get', '/api/training-devices');
+    const devices = useMemo(() => availableDevices?.devices ?? [], [availableDevices]);
+    const hasDevices = devices.length > 0;
+    const isLoadingDevices = isLoading || isRefetching;
+
+    useEffect(() => {
```

Comment:
I think we could avoid this `useEffect` by moving the query (using suspense query) to the location where we define the `useState` for the selected device, and then use the query's data to select the default device.

Something like this,
```typescript
    const {
        data: availableDevices = [],
    } = $api.useQuery('get', '/api/training-devices');

    const [selectedDevice, setSelectedDevice] = useState<string | null>(availableDevices.at(0) ?? null);
```